### ISSUE TITLE
type unstable broadcast fails on master

### ISSUE THREAD

**Issue by piever** [JuliaLang/julia#38460]
I've just noticed that a specific type-unstable broadcast fails on master, while it works fine on julia 1.5

MWE:

on julia 1.5

```julia
julia> broadcast((args...) -> NamedTuple{(:a, :b)}(args), [sin, cos], [1, 2])
2-element Array{NamedTuple{(:a, :b),T} where T<:Tuple,1}:
 (a = sin, b = 1)
 (a = cos, b = 2)
```

on nightly build

```julia
julia> broadcast((args...) -> NamedTuple{(:a, :b)}(args), [sin, cos], [1, 2])
ERROR: TypeError: in typeassert, expected AbstractVector{var"#s825"} where var"#s825"<:(NamedTuple{(:a, :b), _A} where _A<:Tuple{Function, Int64}), got a value of type Vector{NamedTuple{(:a, :b), T} where T<:Tuple}
Stacktrace:
 [1] copy
   @ ./broadcast.jl:928 [inlined]
 [2] materialize
   @ ./broadcast.jl:881 [inlined]
 [3] broadcast(::var"#1#2", ::Vector{Function}, ::Vector{Int64})
   @ Base.Broadcast ./broadcast.jl:819
 [4] top-level scope
   @ REPL[1]:1
```

Here's my versioninfo

```julia
julia> versioninfo()
Julia Version 1.6.0-DEV.1493
Commit f3252bf505 (2020-11-14 16:56 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.0 (ORCJIT, skylake)
```

(I've just redownloaded the nightly build, banner says 2 days old master.)

**Comment by quinnj** [comment]
@nalimilan ^^

**Comment by nalimilan** [comment]
Looks like a duplicate of https://github.com/JuliaLang/julia/issues/38422.

**Comment by piever** [comment]
Ups! Somehow I missed #38422... Yes, closing this one since it's a duplicate and the MWE there seems cleaner. (It's probably worthwhile to check that this is also fixed by #38428, but from what I understand it would just return a `Vector{Any}` with the correct content.)

**Comment by nalimilan** [comment]
I confirm it's fixed too.
